daemon/create.go

```diff
@@ -106,6 +106,10 @@ func (daemon *Daemon) create(params *ContainerCreateConfig) (retC *container.Con
 		return nil, err
 	}
 
+	if err := daemon.updateContainerNetworkSettings(container); err != nil {
```

Comment:
This method must be called only the first time and that is the reason why it was originally protected with
`if len(container.NetworkSettings.Networks) == 0 {` 
in the allocateNetwork function.

Without that protection, a container restarted with multiple networks will start to behave incorrectly (especially ungraceful daemon restart cases). I guess we need to add the above protection (``if len(container.NetworkSettings.Networks) == 0 {` `) here as well.